messgen/ts_generator.py

```diff
@@ -12,150 +16,166 @@
     "int64": "bigint",
     "uint64": "bigint",
     "float32": "number",
-    "float64": "bigint",
+    "float64": "number",
     "string": "string",
+    "bytes": "Uint8Array",
 }
 
-ts_types_array = {
-    "int8": "Int8Array",
-    "uint8": "Uint8Array",
-    "int16": "Int16Array",
-    "uint16": "Uint16Array",
-    "int32": "Int32Array",
-    "uint32": "Uint32Array",
-    "int64": "BigInt64Array",
-    "uint64": "BigUint64Array",
-    "float32": "Float32Array",
-    "float64": "Float64Array",
-}
-
-
-def to_camelcase(str):
-    return ''.join(x for x in str.title().replace('_', '') if not x.isspace())
-
-
-def format_type(f, use_typed_arrays=False):
-    t = ts_types_map.get(f["type"], f["type"])
-    f_type = t[0].lower() + t[1:]
+class TypeScriptGenerator:
+    def __init__(self, protos: Protocols, options: dict):
+        self.protos = protos
+        self.options = options
+        self.generated_types = {}
+        self.imports = set()
+        self.code_lines = []
+
+    def to_camel_case(self, s):
```

Comment:
How about 
 
```suggestion
    def _to_camel_case(self, s: str) -> str:
```

* leading `_` means private by convention
* the type hints are useful for correctness checks with mypy